Dashboard/Controllers/BuildsController.cs

```diff
@@ -101,6 +101,105 @@ public ActionResult View(bool pr = false, DateTimeOffset? startDate = null, stri
             return View(viewName: "View", model: model);
         }
 
+        /// <summary>
+        /// A view of the elapsed time grouped by the result.
+        /// </summary>
+        /// <returns></returns>
+        public ActionResult ElapsedTime(bool pr = false, DateTimeOffset? startDate = null, string viewName = AzureUtil.ViewNameRoslyn)
+        {
+            var filter = CreateBuildFilter(actionName: nameof(ElapsedTime), viewName: viewName, startDate: startDate, pr: pr);
+            var results =
+                _buildUtil.GetBuildResults(filter.StartDate, viewName)
+                .Where(x => pr || !JobUtil.IsPullRequestJobName(x.JobId))
+                .ToList();
+
+            var totalCount = results.Count;
+            var totalSucceeded = results.Count(x => x.ClassificationKind == ClassificationKind.Succeeded);
+
+            var runCounts = results
+                .Select(x => new ElapsedTimeModel() { JobId = x.JobId, JobName = x.JobName, ElapsedTime = x.DurationSeconds })
+                .ToList();
+
+            List<int> runsPerETRange = new List<int>();
+
+            for (int i = 0; i < 6; i++)
+            {
+                runsPerETRange.Add(0);
+            }
+
+            foreach (var runElapsedTime in runCounts)
+            {
+                int ETDigits = runElapsedTime.ElapsedTime.ToString().Length;
+                runsPerETRange[ETDigits - 1] = runsPerETRange[ETDigits - 1] + 1;
+            }
+
+            var model = new ElapsedTimeSummaryModel()
+            {
+                Filter = filter,
+                TotalBuildCount = totalCount,
+                TotalSucceededCount = totalSucceeded,
+                RunCountsPerETRange = runsPerETRange
+            };
+
+            return View(viewName: "ElapsedTime", model: model);
+        }
+
+        /// <summary>
+        /// A view of the total elapsed time per team/project repo, ranked from most elapsed time to least.
+        /// </summary>
+        /// <returns></returns>
+        public ActionResult RepoET(bool pr = false, DateTimeOffset? startDate = null)
```

Comment:
Sure. Original name means "Git repository elapsed time". As it's confusing, I've changed it to "ProjectElapsedTime"